Explorer/Assets/DCL/Character/CharacterCamera/Components/CameraComponent.cs

```diff
@@ -8,13 +9,22 @@ namespace DCL.CharacterCamera
     /// </summary>
     public struct CameraComponent
     {
-        public readonly Camera Camera;
-
         public CameraMode Mode;
+        public int CameraInputLocks;
+        public readonly Camera Camera;
 
         public CameraComponent(Camera camera) : this()
         {
             Camera = camera;
+            CameraInputLocks = 0;
         }
+
+        public bool CameraInputChangeEnabled => CameraInputLocks == 0;
+
+        public void AddCameraInputLock() =>
+            CameraInputLocks++;
+
+        public void RemoveCameraInputLock() =>
+            CameraInputLocks = CameraInputLocks - 1 < 0 ? 0 : CameraInputLocks - 1;
```

Comment:
Good point, I made its setter private at https://github.com/decentraland/unity-explorer/pull/340 since its value needs to be read [at CameraModeAreaHandlerSystem](https://github.com/decentraland/unity-explorer/blob/8ad95c5f9b59267c9e6755889184535a872ed4e7/Explorer/Assets/DCL/SDKComponents/CameraModeArea/Systems/CameraModeAreaHandlerSystem.cs#L77)